lib/puppet/provider/gnupg_key/gnupg.rb

```diff
@@ -73,9 +76,9 @@ def add_key_from_key_content
     path = create_temporary_file(user_id, resource[:key_content])
     command = "gpg --import #{path}"
     begin
-      output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
+    output = Puppet::Util::Execution.execute(command, :uid => user_id, :failonfail => true)
     rescue Puppet::ExecutionFailure => e
-      raise Puppet::Error, "Error while importing key #{resource[:key_id]} using key content:\n#{output}}"
+      raise Puppet::Error, "Error while importing key #{resource[:key_id]} using key content: #{output}"
```

Comment:
If puppet raises an exception during the execute here, the `output` variable isn't going to get set.

This means that when you raise a `Puppet::Error` in the rescue-block, the `output` variable is going to be `nil`. That won't lead to an unhandled exception, but it means Ruby will just interpolate nothing into the error string.

I think you want:

```ruby
raise Puppet::Error, "Error while importing key #{resource[:key_id]} using key content: #{resource[:key_content]}"
```